tdrs-backend/tdpservice/parsers/case_consistency_validator.py

```diff
@@ -77,49 +121,212 @@ def validate(self):
+                for record, schema in t2s:
+                    self.__generate_and_add_error(
+                        schema,
+                        record,
+                        field='RPT_MONTH_YEAR',
+                        msg=(
+                            f'Every {t2_model_name} record should have at least one corresponding '
+                            f'{t1_model_name} record with the same RPT_MONTH_YEAR and CASE_NUMBER.'
+                        )
+                    )
+                    num_errors += 1
+
+                for record, schema in t3s:
+                    self.__generate_and_add_error(
+                        schema,
+                        record,
+                        field='RPT_MONTH_YEAR',
+                        msg=(
+                            f'Every {t3_model_name} record should have at least one corresponding '
+                            f'{t1_model_name} record with the same RPT_MONTH_YEAR and CASE_NUMBER.'
+                        )
+                    )
+                    num_errors += 1
+
+        return num_errors
+
+    def __validate_s2_records_are_related(self):
+        """
+        Validate section 2 records are related.
+
+        Every T4 record should have at least one corresponding T5 record
+        with the same RPT_MONTH_YEAR and CASE_NUMBER.
+        """
+        num_errors = 0
+        is_ssp = self.program_type == 'SSP'
+
+        t4_model_name = 'M4' if is_ssp else 'T4'
+        t4_model = self.__get_model(t4_model_name)
+        t5_model_name = 'M5' if is_ssp else 'T5'
+        t5_model = self.__get_model(t5_model_name)
+
+        logger.debug('validating records are related')
+        logger.debug(f'program type: {self.program_type}')
+        logger.debug(f'section: {self.section}')
+        logger.debug(f'is_ssp: {is_ssp}')
+        logger.debug(f'models - T4: {t4_model}; T5: {t5_model};')
+
+        cases = self.record_schema_pairs.sorted_cases
+        logger.debug(f'cases obj: {cases}')
+
+        for rpt_month_year, reporting_year_cases in cases.items():
+            t4s = reporting_year_cases.get(t4_model, [])
+            t5s = reporting_year_cases.get(t5_model, [])
+
+            logger.debug(f't4s: {t4s}')
+            logger.debug(f't5s: {t5s}')
+
+            if len(t4s) > 0:
+                if len(t5s) == 0:
+                    logger.debug('no t5s')
+                    for record, schema in t4s:
+                        self.__generate_and_add_error(
+                            schema,
+                            record,
+                            field='RPT_MONTH_YEAR',
+                            msg=(
+                                f'Every {t4_model_name} record should have at least one corresponding '
+                                f'{t5_model_name} record with the same RPT_MONTH_YEAR and CASE_NUMBER.'
+                            )
+                        )
+                        num_errors += 1
+                else:
```

Comment:
Can we omit this empty `else` since it doesnt do anything?

Reply:
i won't die on this hill but i do like it for clarity on the success case

Reply:
Thats fine with me also.